grpc-bridge/src/main/java/org/jboss/resteasy/grpc/protobuf/JavaToProtobufGenerator.java

```diff
@@ -0,0 +1,979 @@
+        .append("   string path = ").append(counter++).append(";\n")
+        .append("   string domain = ").append(counter++).append(";\n")
+        .append("}");
+      sb.append("\n\nmessage ServletInfo {\n")
+        .append("   string characterEncoding = ").append(counter++).append(";\n")
+        .append("   string clientAddress = ").append(counter++).append(";\n")
+        .append("   string clientHost = ").append(counter++).append(";\n")
+        .append("   int32  clientPort = ").append(counter++).append(";\n")
+        .append("}");
+      sb.append("\n\nmessage FormValues {\n")
+        .append("   repeated string formValues_field = ").append(counter++).append(";\n")
+        .append("}");
+      sb.append("\n\nmessage FormMap {\n")
+        .append("   map<string, FormValues> formMap_field = ").append(counter++).append(";\n")
+        .append("}");
+      sb.append("\n\nmessage GeneralEntityMessage {\n")
+        .append("   ServletInfo servletInfo = ").append(counter++).append(";\n")
+        .append("   string URL = ").append(counter++).append(";\n")
+        .append("   map<string, gHeader> headers = ").append(counter++).append(";\n")
+        .append("   repeated gCookie cookies = ").append(counter++).append(";\n")
+        .append("   oneof messageType {\n");
+      for (String messageType : entityMessageTypes) {
+         sb.append("      ")
+         .append(messageType)
+         .append(" ")
+         .append(messageType).append("_field")
+         .append(" = ")
+         .append(counter++)
+         .append(";\n");
+      }
+    sb.append("      FormMap form_field = ").append(counter++).append(";\n");
+    sb.append("   }\n}\n");
+   }
+
+   private static void createGeneralReturnMessageType(StringBuilder sb) {
+      sb.append("\nmessage GeneralReturnMessage {\n")
+        .append("   map<string, gHeader> headers = ").append(counter++).append(";\n")
+        .append("   repeated gCookie cookies = ").append(counter++).append(";\n")
+        .append("   oneof messageType {\n");
+    for (String messageType : returnMessageTypes) {
+       sb.append("      ")
+         .append(messageType)
+         .append(" ")
+         .append(namify(messageType)).append("_field")
+         .append(" = ")
+         .append(counter++)
+         .append(";\n");
+    }
+    sb.append("   }\n}\n");
+   }
+
+   private static void writeProtoFile(String[] args, StringBuilder sb) throws IOException {
+      String path = args[0];
+      String generatedSources = "src/main/proto";
+      for (String s : generatedSources.split("/")) {
+         path += "/" + s;
+         File dir = new File(path);
+         if(!dir.exists()){
+            dir.mkdir();
+         }
+      }
+      File file = new File(path + "/" + args[3] + ".proto");
+      file.createNewFile();
+      FileWriter fw = new FileWriter(file.getAbsoluteFile());
+      BufferedWriter bw = new BufferedWriter(fw);
+      bw.write(sb.toString());
+      bw.close();
```

Comment:
This could be something like:
```java
final Path file = Paths.of(path, args[3] + ".proto");
Files.writeString(file, sb, StandardCharset.UTF_8);
```

It should at least use try-with-resources and close both streams.

Reply:
Ok, I've got

      final Path path = Path.of(args[0], "src", "main", "proto", args[3] + ".proto");
      Files.writeString(path, sb.toString(), StandardCharsets.UTF_8);